VisualizationBase/src/overlays/ArrowOverlay.cpp

```diff
@@ -34,12 +35,28 @@ DEFINE_ITEM_COMMON(ArrowOverlay, "item")
 ArrowOverlay::ArrowOverlay(Item* arrowFrom, Item* arrowTo, const StyleType* style)
 	: Super{{arrowFrom, arrowTo}, style}
 {
+	setAcceptHoverEvents(true);
+	activateDefaultArrowStyle();
+}
+
+void ArrowOverlay::activateDefaultArrowStyle()
+{
+	currentArrowBrush_ = this->style()->arrowBrush();
```

Comment:
So, instead of storing specific styles in fields, it would be better, to directly select the right style in the `paint` method. Just have a single boolean field `selected_` that controls this. Then you don't need this and the next methods.